fitparser/src/de/parser.rs

```diff
@@ -225,97 +225,41 @@ impl FitDataMessage {
         self.time_offset
     }
 
-    /// Data field mapping of <field_number, Value>
-    pub fn fields(&self) -> &HashMap<u8, Value> {
+    /// Data field mapping of <(dev_data_idx, field_number), Value>
+    pub fn fields(&self) -> &HashMap<(u8, u8), Value> {
         &self.fields
     }
 
-    /// Mutable Data field mapping of <field_number, Value>
-    pub fn fields_mut(&mut self) -> &mut HashMap<u8, Value> {
+    /// Mutable Data field mapping of <(dev_data_idx, field_number), Value>
+    pub fn fields_mut(&mut self) -> &mut HashMap<(u8, u8), Value> {
         &mut self.fields
     }
-
-    #[allow(dead_code)]
-    /// Developer field data
-    pub fn developer_fields(&self) -> &[Value] {
```

Comment:
As above I think we want to keep dev fields separate from the regular set of fields, so we might need a mutable version of this?

Reply:
I have pushed a commit that keeps the normal fields separate from the developer fields. There is an issue with the borrow checker though: We can't use getter functions like the above linked one to get both a mutable reference to fields *and* a non mutable reference to developer_fields, which we would need for our call to `mesg_num.decode_message`.
To work around this issue I made fields and developer_fields public, to be able to directly borrow them without going through message.